teuthology/task/internal/lock_machines.py

```diff
@@ -23,6 +23,14 @@ def lock_machines(ctx, config):
     new machines.  This is not called if the one has teuthology-locked
     machines and placed those keys in the Targets section of a yaml file.
     """
+    lock_machines_helper(ctx, config)
+    try:
+        yield
+    finally:
+        unlock_machines(ctx)
+
+
+def lock_machines_helper(ctx, config, reimage=True):
```

Comment:
I think we decided to not touch internal tasks for now. The dispatcher should not use lock_machines module in any ways.

Reply:
It doesn't change the working of the internal task. Just refactors it such that the same logic can be used in dispatcher to wait to for required number of machines to be available and then lock them (using `lock_many` of lock module). Since the waiting logic is not in the lock module, I thought it's best to reuse the logic from here to avoid repetitive code and still maintaining backward compatibility. Please let me know if this needs to be done differently.

Reply:
I can see it does not change the behavior of the code, it just wrong place for this can of refactoring right now, overall locking mechanism should be rewritten in future, but for the goals of this PR those changes are not required. I would prefer to leave internal tasks untouched right now. Later we should refactor the lock_machines tasks to use teuthology.lock.ops code. The dispatcher should not user anything from the directory teuthology/task, and use lock api directly. The tasks should only be invoked from the run itself.

Reply:
Added func [`block_and_lock_machines`](https://github.com/ceph/teuthology/pull/1546/files#diff-0c1bc86b25e221cf13dbc1b8dee22354R318) in lock module which is used by both the `lock_machines` task and dispatcher. Is this approach okay?